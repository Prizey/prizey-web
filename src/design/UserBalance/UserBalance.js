import React from 'react'
import get from 'lodash/get'
import { connect } from 'react-redux'
import { withStyles } from '@material-ui/core/styles'
import { Typography } from '@material-ui/core'

const styles = theme => ({
  icon: {
    height: theme.spacing.md,
    marginRight: theme.spacing.xs,
  },
  root: {
    alignItems: 'center',
    display: 'flex',
    [theme.breakpoints.down('xs')]: {
      marginTop: 0,
    },
  },
})

export const UserBalanceComponent = withStyles(styles)(({ classes, user }) => (
  <div className={classes.root}>
    <img src="/icons/diamond.png" alt="diamond" className={classes.icon} />
    <Typography>{user ? user.tickets : '0'}</Typography>
  </div>
))

export const mapState = ({ auth }) => ({
  user: get(auth, 'currentUser.0.state.info'),
})

export default connect(
  mapState,
  null,
)(UserBalanceComponent)
