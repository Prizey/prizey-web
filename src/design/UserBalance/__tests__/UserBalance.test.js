import React from 'react'
import renderer from 'react-test-renderer'
import { UserBalanceComponent, mapState } from '../UserBalance'

it('renders correctly', () => {
  const params = {
    user: { tickets: 10 },
  }
  const tree = renderer.create(<UserBalanceComponent {...params} />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('get the user from croods state', () => {
  const user = { id: 1, tickets: 10 }
  const state = {
    auth: {
      currentUser: [{ state: { info: user } }],
    },
  }

  const result = mapState(state)
  expect(result).toEqual({ user })
})
