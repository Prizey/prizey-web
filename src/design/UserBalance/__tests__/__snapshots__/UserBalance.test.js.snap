// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Component-root-2"
>
  <img
    alt="diamond"
    className="Component-icon-1"
    src="/icons/diamond.png"
  />
  <p
    className="MuiTypography-root-3 MuiTypography-body1-12"
  >
    10
  </p>
</div>
`;
